src/components/TextInput/TextInput.scss

```diff
@@ -0,0 +1,112 @@
+@use '~@quartz/styles/scss/borders';
+@use '~@quartz/styles/scss/color-scheme';
+@use '~@quartz/styles/scss/fonts';
+@use '~@quartz/styles/scss/helpers/resets';
+@use '~@quartz/styles/scss/tokens';
+
+.container {
+	position: relative;
+	border: 0;
+	padding: 0;
+	margin: 0;
+}
+
+.label {
+	@include fonts.maison-extended-700-1;
+
+	text-transform: uppercase;
+	color: color-scheme.$typography;
+	display: block;
+	line-height: 1.25;
+
+	&.required {
+		display: flex;
+		flex-wrap: wrap;
+
+		&::after {
+			content: '*';
+
+			@include fonts.maison-800-3;
+
+			color: color-scheme.$accent;
+			padding: 0 4px;
+			line-height: 0.8;
+		}
+	}
+}
+
+.input {
+	@include resets.text-input;
+	@include fonts.maison-500-3;
+
+	position: relative;
+	display: inline-block;
+	height: 44px;
+	width: 100%;
+	border-radius: tokens.$size-border-radius;
+	border: borders.$solid-interactive;
+	box-sizing: border-box;
+	font-size: 16px;
+	font-weight: 300;
+	color: color-scheme.$typography;
+	text-indent: 12px;
+	background-color: color-scheme.$background-2;
+	padding: 0;
+	transition: border-color, 0.25s;
+	margin: 8px 0;
+
+	&::placeholder {
+		color: color-scheme.$typography-faint;
+	}
+
+	&:hover,
+	&:focus {
+		border-color: color-scheme.$accent;
+	}
+}
+
+.textarea {
+	@include resets.text-input;
+	@include fonts.maison-500-3;
+
+	resize: none;
+	box-sizing: border-box;
+	display: block;
+	width: 100%;
+	font-size: 16px;
+	line-height: 1.25;
+	color: color-scheme.$typography;
+	border: borders.$solid-interactive;
+	background-color: color-scheme.$background-2;
+	border-radius: tokens.$size-border-radius;
```

Comment:
seems like input and textarea share a lot of properties, consider using a shared selector for those properties to encourage future developers to keep it that way